p384/src/arithmetic/projective.rs

```diff
@@ -594,3 +594,120 @@ impl TryFrom<&ProjectivePoint> for PublicKey {
         AffinePoint::from(point).try_into()
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::{AffinePoint, ProjectivePoint};
+    use crate::test_vectors::ADD_TEST_VECTORS;
+    use elliptic_curve::group::prime::PrimeCurveAffine;
+
+    #[test]
+    fn affine_to_projective() {
+        let basepoint_affine = AffinePoint::GENERATOR;
+        let basepoint_projective = ProjectivePoint::GENERATOR;
+
+        assert_eq!(
+            ProjectivePoint::from(basepoint_affine),
+            basepoint_projective,
+        );
+        assert_eq!(basepoint_projective.to_affine(), basepoint_affine);
+        assert!(!bool::from(basepoint_projective.to_affine().is_identity()));
+
+        // TODO(tarcieri): BROKEN!
+        // assert!(bool::from(
+        //     ProjectivePoint::IDENTITY.to_affine().is_identity()
+        // ));
```

Comment:
@jedisct1 @brycx FYI, this test is failing

Reply:
I think I found the issue: #568